homeassistant/scripts/influxdb_migrator.py

```diff
@@ -0,0 +1,177 @@
+
+
+# Based on code at
+# http://stackoverflow.com/questions/3173320/text-progress-bar-in-the-console
+def print_progress(iteration: int, total: int, prefix: str='', suffix: str='',
+                   decimals: int=2, bar_length: int=68) -> None:
+    """Print progress bar.
+    Call in a loop to create terminal progress bar
+    @params:
+        iteration   - Required  : current iteration (Int)
+        total       - Required  : total iterations (Int)
+        prefix      - Optional  : prefix string (Str)
+        suffix      - Optional  : suffix string (Str)
+        decimals    - Optional  : number of decimals in percent complete (Int)
+        barLength   - Optional  : character length of bar (Int)
+    """
+    filled_length = int(round(bar_length * iteration / float(total)))
+    percents = round(100.00 * (iteration / float(total)), decimals)
+    line = '#' * filled_length + '-' * (bar_length - filled_length)
+    sys.stdout.write('%s [%s] %s%s %s\r' % (prefix, line,
+                                            percents, '%', suffix))
+    sys.stdout.flush()
+    if iteration == total:
+        print("\n")
+
+
+def run(script_args: List) -> int:
+    """The actual script body."""
+    from influxdb import InfluxDBClient
+
+    parser = argparse.ArgumentParser(
+        description="Migrate legacy influxDB.")
+    parser.add_argument(
+        '-d', '--dbname',
+        metavar='dbname',
+        required=True,
+        help="InfluxDB database name")
+    parser.add_argument(
+        '-H', '--host',
+        metavar='host',
+        default='127.0.0.1',
+        help="InfluxDB host address")
+    parser.add_argument(
+        '-P', '--port',
+        metavar='port',
+        default=8086,
+        help="InfluxDB host port")
+    parser.add_argument(
+        '-u', '--username',
+        metavar='username',
+        default='root',
+        help="InfluxDB username")
+    parser.add_argument(
+        '-p', '--password',
+        metavar='password',
+        default='root',
+        help="InfluxDB password")
+    parser.add_argument(
+        '-s', '--step',
+        metavar='step',
+        default=1000,
+        help="How many points are migrate in the same time")
+    parser.add_argument(
+        '-D', '--delete',
+        action='store_true',
+        default=False,
+        help="Delete old database")
+    parser.add_argument(
+        '--script',
+        choices=['influxdb_migrator'])
+
+    args = parser.parse_args()
+
+    # Get client for old DB
+    client = InfluxDBClient(args.host, args.port,
+                            args.username, args.password)
+    client.switch_database(args.dbname)
+    # Get DB list
+    db_list = [db['name'] for db in client.get_list_database()]
+    # Get measurements of the old DB
+    res = client.query('SHOW MEASUREMENTS')
+    measurements = [measurement['name'] for measurement in res.get_points()]
+    nb_measurements = len(measurements)
+    # Move data
+    # Get old DB name
```

Comment:
I think you should clarify that you make a backup db and copy the data from the old db to the backup db. First time I read this part of the script, I was wondering why you copied data from the old db to the old db. :wink: